src/oatpp/web/server/HttpProcessor.cpp

```diff
@@ -244,21 +248,31 @@ void HttpProcessor::Task::run(){
     // DO NOTHING
   }
 
+  (*m_counter)--;
```

Comment:
To my mind, a better place to dec counter is the destructor.

Reply:
Have you read the commit message? That is exactly I was doing in the first place. However, `std::thread` does not reliably clean the `HttpProcessor` on thread exit. It seems that the resources are only cleaned when the ConnectionHandler is cleaned.